### PR TITLE
Ensure body is closed after error is checked

### PR DESCRIPTION
**Pull request by runcom**
Signed-off-by: Antonio Murdaca runcom@linux.com

### CODE REVIEW

File: integration-cli/docker_api_logs_test.go [hunk 1]
```diff
@@ -28,6 +28,7 @@ func (s *DockerSuite) TestLogsApiWithStdout(c *check.C) {
 	go func() {
 		res, body, err := sockRequestRaw("GET", fmt.Sprintf("/containers/%s/logs?follow=1&stdout=1&timestamps=1", id), nil, "")
 		out, _ := bufio.NewReader(body).ReadString('\n')
```
Comment (duglin) [inline integration-cli/docker_api_logs_test.go]: no check of 'err' here needed?

Comment (runcom) [inline integration-cli/docker_api_logs_test.go]: mm yes, this goroutine needs to be rewritten a bit, thanks @duglin

Comment (runcom) [inline integration-cli/docker_api_logs_test.go]: updated PTAL

File: integration-cli/docker_api_logs_test.go [hunk 2]
```diff
@@ -27,7 +27,14 @@ func (s *DockerSuite) TestLogsApiWithStdout(c *check.C) {
...
 		res, body, err := sockRequestRaw("GET", fmt.Sprintf("/containers/%s/logs?follow=1&stdout=1&timestamps=1", id), nil, "")
-		out, _ := bufio.NewReader(body).ReadString('\n')
+		if err != nil {
+			chLog <- logOut{"", nil, err}
+		}
+		out, err := bufio.NewReader(body).ReadString('\n')
+		if err != nil {
+			chLog <- logOut{"", nil, err}
+		}
+		body.Close()
```
Comment (duglin) [inline integration-cli/docker_api_logs_test.go]: should this be in a defer and moved to before line 33?  That way if body isn't nil but something goes wrong with the ReadString() we'll still do body.Close()?

Also, should there be a return after each chLog <- ...  in those error cases?

Comment (runcom) [inline integration-cli/docker_api_logs_test.go]: updated! :)

**Comment by duglin** [comment]
LGTM

**Comment by estesp** [comment]
LGTM

**Pull request merged**